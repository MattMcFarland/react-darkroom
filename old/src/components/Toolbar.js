import React from 'react';

const toList = (node, index) => <li key={index}>{node}</li>;

export default ({children}) =>
  <menu>
    <ul>
      {React.Children.toArray(children).map(toList)}
    </ul>
  </menu>;
