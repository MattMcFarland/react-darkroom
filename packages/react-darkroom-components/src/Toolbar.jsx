import React from 'react';
import injectSheet from 'react-jss';
import PropTypes from 'prop-types';

const Toolbar = ({ classes, children }) =>
  (<menu className={classes.menu}>
    <ul className={classes.list}>
      { React.Children.toArray(children).map((node, index) =>
        <li className={classes.listItem} key={`toolbar-${index + 1}`}>{node}</li>) }
    </ul>
  </menu>);

Toolbar.propTypes = {
  /** @ignore JSS styles */
  classes: PropTypes.shape({
    menu: PropTypes.string,
    list: PropTypes.string,
  }).isRequired,
  /** The buttons which exist in the toolbar, 
   * best results are used in conjunction with `<ToolbarButton/>` but can be anything! */
  children: PropTypes.node,
};

const styles = {
  menu: {
    margin: 0,
    padding: '0 8px',
    background: '#444',
    color: '#eee',
    display: 'inline-block',
    position: 'relative',
    borderRadius: 3,
    fontSize: 20,
    lineHeight: 1,
    '&:before': {
      content: ' ',
      height: 0,
      width: 0,
      position: 'absolute',
      pointerEvents: 'none',
      border: 'solid rgba(255, 255, 255, 0)',
      borderColor: 'rgba(255, 255, 255, 0)',
      borderWidth: 10,
      bottom: -20,
      left: '1em',
      marginLeft: -10,
    },
    '&:after': {
      content: ' ',
      height: 0,
      width: 0,
      position: 'absolute',
      pointerEvents: 'none',
      border: 'solid rgba(255, 255, 255, 0)',
      borderColor: 'rgba(255, 255, 255, 0)',
      borderWidth: 9,
      bottom: -18,
      borderTopColor: '#444',
      left: '1em',
      marginLeft: -9,
    },
  },
  list: {
    listStyleType: 'none',
    padding: 0,
    margin: 0,
    display: 'inline-block',
  },
  listItem: {
    display: 'inline-block',
  },
};

export default injectSheet(styles)(Toolbar);
