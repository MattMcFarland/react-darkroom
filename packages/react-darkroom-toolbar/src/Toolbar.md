Used as a container for pluggable `ToolbarButtons`

    <Toolbar>
      <ToolbarButton>
        File        
      </ToolbarButton>
      <ToolbarButton>
        Edit
      </ToolbarButton>
      <ToolbarButton>
        About
      </ToolbarButton>
      <ToolbarButton>
        <Icon kind="checkmark"/>
      </ToolbarButton>      
    </Toolbar>
