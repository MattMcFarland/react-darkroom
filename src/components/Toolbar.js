import React from 'react';

export default ({
  children
  }) => {

  var
    childNodes = [];

  React.Children.forEach(children, child => {
    childNodes.push(child);
  });

  return (
    <menu>
      <ul>
        {childNodes.map((childNode, index) => {
          return (<li key={index}>{childNode}</li>)
        })}
      </ul>
    </menu>
  );
};
