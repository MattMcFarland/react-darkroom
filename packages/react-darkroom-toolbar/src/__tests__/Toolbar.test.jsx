import React from 'react';
import { shallow } from 'enzyme';

import Toolbar from '../Toolbar';

const { describe, test, expect } = global;

describe('Toolbar', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(<Toolbar><button>test</button></Toolbar>);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
