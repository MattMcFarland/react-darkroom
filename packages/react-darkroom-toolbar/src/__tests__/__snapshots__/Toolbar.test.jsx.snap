// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar matches snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "class": "menu-0-0",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "list-0-1",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "listItem-0-2",
                },
                "children": Array [
                  Object {
                    "attribs": Object {},
                    "children": Array [
                      Object {
                        "data": "test",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "button",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                  },
                ],
                "name": "li",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
              },
            ],
            "name": "ul",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
          },
        ],
        "name": "menu",
        "next": null,
        "parent": null,
        "prev": null,
        "root": [Circular],
        "type": "tag",
      },
    ],
    "name": "root",
    "next": null,
    "parent": null,
    "prev": null,
    "type": "root",
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xmlMode": false,
  },
}
`;
